lib/virt_autotest/utils.pm

```diff
@@ -108,13 +111,27 @@ sub ssh_setup {
 }
 
 sub ssh_copy_id {
-    my $guest           = shift;
+    my ($guest, %args) = @_;
+
+    my $username        = $args{username}        // 'root';
+    my $authorized_keys = $args{authorized_keys} // '.ssh/authorized_keys';
+    my $scp             = $args{scp}             // 0;
     my $mode            = is_sle('=11-sp4')             ? ''                      : '-f';
     my $default_ssh_key = (!(get_var('VIRT_AUTOTEST'))) ? "/root/.ssh/id_rsa.pub" : "/var/testvirt.net/.ssh/id_rsa.pub";
     script_retry "nmap $guest -PN -p ssh | grep open", delay => 15, retry => 12;
     assert_script_run "ssh-keyscan $guest >> ~/.ssh/known_hosts";
-    if (script_run("ssh -o PreferredAuthentications=publickey root\@$guest hostname -f") != 0) {
-        exec_and_insert_password("ssh-copy-id -i $default_ssh_key -o StrictHostKeyChecking=no $mode root\@$guest");
+    if (script_run("ssh -o PreferredAuthentications=publickey -o ControlMaster=no $username\@$guest hostname") != 0) {
+        unless ($scp == 1) {
+            exec_and_insert_password("ssh-copy-id -i $default_ssh_key -o PreferredAuthentications=password,keyboard-interactive -o ControlMaster=no $mode $username\@$guest");
+        } else {
+            exec_and_insert_password("ssh -o PreferredAuthentications=password,keyboard-interactive -o ControlMaster=no $username\@$guest 'mkdir .ssh' || true");
+            exec_and_insert_password("scp -o PreferredAuthentications=password,keyboard-interactive -o ControlMaster=no $default_ssh_key $username\@$guest:'$authorized_keys'");
```

Comment:
If we're manually pushing the `authorized_keys`, we should also ensure the permissions are set correctly as well:

    exec_and_insert_password("ssh -o PreferredAuthentications=password,keyboard-interactive -o ControlMaster=no $username\@$guest 'chmod 0700 /root/.ssh'");
    exec_and_insert_password("ssh -o PreferredAuthentications=password,keyboard-interactive -o ControlMaster=no $username\@$guest ''chmod 0644 /root/.ssh/authorized_keys");

Reply:
👍 thank you, I did:
```perl
exec_and_insert_password("ssh -o PreferredAuthentications=password,keyboard-interactive -o ControlMaster=no $username\@$guest 'chmod 0700 ~/.ssh/'");
exec_and_insert_password("ssh -o PreferredAuthentications=password,keyboard-interactive -o ControlMaster=no $username\@$guest 'chmod 0644 ~/.ssh/authorized_keys'");
```